### PR TITLE
New deb822_repository module

### PR DESCRIPTION
**Pull request by sivel**
##### ISSUE TYPE
- Feature Pull Request

##### ADDITIONAL INFORMATION

Maybe To Do:

- [x] state
- [x] checkmode
- [x] add test that installs from new repo

References:
- https://discourse.ubuntu.com/t/spec-apt-deb822-sources-by-default/29333
- https://twitter.com/debian/status/1626588286033760256
- https://manpages.debian.org/testing/apt/sources.list.5.en.html
- https://repolib.readthedocs.io/en/latest/deb822-format.html

### CODE REVIEW

**Comment by sivel** [comment]
Need to evaluate module respawn, but it does look like this is available in pypi https://pypi.org/project/python-debian/

Need to confirm whether it's being actively updated.

**Comment by sivel** [comment]
Ok, that didn't take much time, looks like I see `0.1.36` in ubuntu 22.04, and pypi is actually a little advanced, up to `0.1.49`, but in any case, the dep should be resolvable regardless of Python version in use on the target.  Also, the use of `dist-packages` on debian based distros probably allows for it as well.

**Review by Conan-Kudo — APPROVED** [review]

**Comment by sivel** [comment]
I'm so sorry the bot keeps spamming people!  I'll try to get that under control

bot_skip

**Review by bcoca — APPROVED** [review]

File: changelogs/fragments/deb822-repository.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- deb822_repository - Add new module for managing DEB822 formatted apt repositories
```
Comment (Andersson007) [inline changelogs/fragments/deb822-repository.yml:2]:
```suggestion
- deb822_repository - Add new module for managing DEB822 formatted apt repositories (https://github.com/ansible/ansible/pull/80018).
```

File: lib/ansible/modules/deb822_repository.py [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+            try:
+                key_changed, signed_by_filename, signed_by_data = write_signed_by_key(module, value, slug)
+                value = signed_by_filename or signed_by_data
+                changed |= key_changed
+            except RuntimeError as exc:
+                module.fail_json(
+                    msg='Could not fetch signed_by key: %s' % to_native(exc)
+                )
+
+        if value.count('\n') > 0:
```
Comment (s-hertel) [inline lib/ansible/modules/deb822_repository.py:523]: I think this is missing a condition for type int - `date_max_future` hasn't been popped out of params and would give a SyntaxError here

**Review by s-hertel — COMMENTED** [review]
This seems nice to have, lgtm.

File: lib/ansible/modules/deb822_repository.py (hunk 2 above)
Comment (sivel) [reply lib/ansible/modules/deb822_repository.py:523]: doh

Comment (sivel) [reply lib/ansible/modules/deb822_repository.py:523]: Fixed

**Review by s-hertel — APPROVED** [review]

**Pull request merged**